### PR TITLE
use to NIO 2 (from master branch) and Swift 5

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

SwiftPM from Swift 5.0 brings targets that only support certain systems
which is really handy for this package as it only support macOS 10.14+
and iOS/tvOS 12+

Modifications:

- made use of the Swift 5.0 manifests which can have restrictions on the
  supported platforms
- adapted to NIO 2

Result:

- better development story
- using the latest & greatest

### CODE REVIEW

File: Sources/NIOTransportServices/SocketAddress+NWEndpoint.swift [hunk 1]
```diff
@@ -98,6 +98,10 @@ internal extension SocketAddress {
             preconditionFailure("Cannot represent service addresses in SocketAddress")
         case .hostPort(.name, _):
             preconditionFailure("Cannot represent host by name only as SocketAddress")
+        case .hostPort(_, _):
```
Comment (Lukasa) [inline Sources/NIOTransportServices/SocketAddress+NWEndpoint.swift]: This case supersedes the one above.

Comment (weissi) [reply Sources/NIOTransportServices/SocketAddress+NWEndpoint.swift]: done

File: Sources/NIOTransportServices/NIOTSEventLoopGroup.swift [hunk 2]
```diff
@@ -77,4 +77,8 @@ public final class NIOTSEventLoopGroup: EventLoopGroup {
             callback(error)
         }
     }
+
+    public func makeIterator() -> EventLoopIterator {
+        return EventLoopIterator(self.eventLoops)
```
Comment (weissi) [inline Sources/NIOTransportServices/NIOTSEventLoopGroup.swift]: needs https://github.com/apple/swift-nio/pull/706 to compile

File: README.md [hunk 3]
```diff
@@ -24,16 +24,10 @@ Today, the easiest way to use SwiftNIO Transport Services is through CocoaPods:
 You can also use the Swift Package Manager, however Network.framework is only available on macOS 10.14+, iOS 12+, and tvOS 12+. This does not match the current minimum deployment target for Swift Package Manager, so building this repository with Swift Package Manager requires that you pass custom build flags, e.g:
 
 ```
-swift build -Xswiftc -target -Xswiftc x86_64-apple-macosx10.14
```
Comment (neonichu) [inline README.md]: 🎉

**Comment by dmbryson** [comment]
Can this move forward?

File: Sources/NIOTransportServices/SocketAddress+NWEndpoint.swift [hunk 4]
```diff
@@ -96,8 +96,10 @@ internal extension SocketAddress {
             self = try .init(unixDomainSocketPath: path)
         case .service:
             preconditionFailure("Cannot represent service addresses in SocketAddress")
-        case .hostPort(.name, _):
+        case .hostPort(_, _):
             preconditionFailure("Cannot represent host by name only as SocketAddress")
```
Comment (Lukasa) [inline Sources/NIOTransportServices/SocketAddress+NWEndpoint.swift]: I don't think this error message is right anymore.

**Comment by ddunbar** [comment]
Any update here?

**Comment by Lukasa** [comment]
I'll bully @weissi into updating this patch today.

**Comment by weissi** [comment]
sorry @ddunbar & @Lukasa for the delay, this week and next I'm super busy but I'll review this rn

**Comment by weissi** [comment]
@Lukasa ready now and sorry again

**Pull request merged**